### ISSUE TITLE
var-annotated error not showing up unless --cache-fine-grained is used

### ISSUE THREAD

**Issue by pjakubo86** [python/mypy#14732]
**Bug Report**

mypy produces a set of `var-annotated` errors when running with `--cache-fine-grained` that it does not when run plainly. As far as I can tell, mypy's documentation doesn't indicate that there should be any difference in the set of errors returned between the two caching modes.

I came across this because I noticed that despite `mypy .` returning no errors when run at the command line, VSCode was highlighting these `var-annotated` errors, which seem real enough. Eventually, I figured out that VSCode is using `dmypy` in the background, which uses the fine-grained cache and I was able to reproduce the errors by using `--cache-fine-grained` at the command line.

**To Reproduce**

1. Create a file `main.py`:
```python
foo = None
```

2. Run `mypy main.py`
3. Run `mypy --cache-fine-grained main.py`

**Expected Behavior**

The output of steps 2 and 3 are the same.

**Actual Behavior**

Step 2 output:
```
% mypy main.py 
Success: no issues found in 1 source file
```

Step 3 output:
```
% mypy --cache-fine-grained main.py
main.py:1: error: Need type annotation for "foo" (hint: "foo: <type> | None = ...")  [var-annotated]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.991, 1.0.0, 1.0.1
- Mypy command-line flags: See above
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9, 3.10

**Comment by hauntsaninja** [comment]
Yeah, this is https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-local-partial-types

> Note: this option is always implicitly enabled in mypy daemon 

See also #10169

**Comment by pjakubo86** [comment]
Thanks, Shantanu! I appreciate the quick turnaround.

I _might_ have found that documentation if it included either the generated error (`var-annotated`) or a reference to the fine-grained cache. I'll look at submitting a PR to improve the documentation a bit.

**Comment by hauntsaninja** [comment]
PR improving the docs most welcome :-)
